src/main/scala/devices/gpio/GPIOPins.scala

```diff
@@ -9,17 +9,11 @@ import sifive.blocks.devices.pinctrl.{Pin}
 // even though it looks like something that more directly talks to
 // a pin. It also makes it possible to change the exact
 // type of pad this connects to.
-class GPIOSignals[T <: Data] (pingen: ()=> T,  c: GPIOParams) extends Bundle {
+class GPIOSignals[T <: Data](private val pingen: ()=> T, val c: GPIOParams) extends Bundle {
```

Comment:
why are some private and some not private?

Reply:
It was an arbitrary choice. `GPIOParams` seems like the kind of thing you *might* want to access externally to see what parameters this Bundle was constructed with. `pingen` on the other hand felt more private to me.

For autoclonetype to work, the arguments merely must be vals, public or private. If you think they should all be public or private I'm happy to make the change.